### ISSUE TITLE
HealthCheck bypasses FaultInjection

### ISSUE THREAD

**Issue by slonka** [kumahq/kuma#5212]
### What happened?

Steps to reproduce:

1. Deploy [kuma-demo](https://github.com/kumahq/kuma-demo/)
2. Remove default cb & retry

```bash
kubectl delete circuitbreakers.kuma.io circuit-breaker-all-default
kubectl delete retries.kuma.io retry-all-default
```

3. Apply Fault Injection

```yaml
apiVersion: kuma.io/v1alpha1
kind: FaultInjection
mesh: default
metadata:
  name: fi1
spec:
  sources:
    - match:
        kuma.io/service: frontend_kuma-demo_svc_8080
        kuma.io/protocol: http
  destinations:
    - match:
        kuma.io/service: backend_kuma-demo_svc_3001
        kuma.io/protocol: http
  conf:        
    abort:
      httpStatus: 500
      percentage: 100
```

4. Apply Health Check

```yaml
apiVersion: kuma.io/v1alpha1
kind: HealthCheck
mesh: default
metadata:
  name: frontend-to-backend
spec:
  sources:
  - match:
      kuma.io/service: frontend_kuma-demo_svc_8080
  destinations:
  - match:
      kuma.io/service: backend_kuma-demo_svc_3001
  conf:
    interval: 1s
    timeout: 2s
    unhealthyThreshold: 1
    healthyThreshold: 10
```

5. Go to stats 

```text
cluster.backend_kuma-demo_svc_3001.health_check.failure: 0
```

HC failure will be 0.

This is because we match for `kuma-tags` to match regex `*&kuma.io/protocol=[^&]*http[,&].*&kuma.io/service=[^&]*frontend_kuma-demo_svc_8080[,&].*` and a HC request will not have those headers.

Part of config dump:

```json
                          {
                            "name": "envoy.filters.http.fault",
                            "typedConfig": {
                              "@type": "type.googleapis.com/envoy.extensions.filters.http.fault.v3.HTTPFault",
                              "abort": {
                                "httpStatus": 500,
                                "percentage": {
                                  "numerator": 100
                                }
                              },
                              "headers": [
                                {
                                  "name": "x-kuma-tags",
                                  "safeRegexMatch": {
                                    "googleRe2": {},
                                    "regex": ".*&kuma.io/protocol=[^&]*http[,&].*&kuma.io/service=[^&]*frontend_kuma-demo_svc_8080[,&].*"
                                  }
                                }
                              ]
                            }
                          }
```

To make the HC work with the FI we need to add that header manually:

```yaml
apiVersion: kuma.io/v1alpha1
kind: HealthCheck
mesh: default
metadata:
  name: frontend-to-backend
spec:
  sources:
  - match:
      kuma.io/service: frontend_kuma-demo_svc_8080
  destinations:
  - match:
      kuma.io/service: backend_kuma-demo_svc_3001
  conf:
    interval: 1s
    timeout: 2s
    unhealthyThreshold: 1
    healthyThreshold: 10
    eventLogPath: "/tmp/events.txt"
    alwaysLogHealthCheckFailures: true
    http:
      path: /
      expectedStatuses: [200]
      requestHeadersToAdd:
      - append: false
        header:
          key: x-kuma-tags
          value: "&kuma.io/protocol=http&&kuma.io/service=frontend_kuma-demo_svc_8080&"
```

I think that nothing should be able to bypass FI because in a real world scenario that bypass would not happened (and FI is there to simulate real world scenarios).

AC:
- make HC go through Fault Injection

**Comment by jakubdyszkiewicz** [comment]
Related to https://github.com/kumahq/kuma/issues/1256

**Comment by jakubdyszkiewicz** [comment]
Will be fixed in Fault Injection 2.0
